File: java01t/src/step01/Test06.java
package step01;

// 블록
// - 명령어를 묶는 문법 
// - 사용처: 조건문, 반복문, 메서드, 클래스 
// - 문법: 중괄호 사용 {}

public class Test06 {

  public static void main(String[] args) {
    // 블록과 변수의 관계
    // - 변수는 자신이 소속된 블록 안에서만 유효하다.
    // - 블록을 벗어나면 그 블록에 선언된 모든 변수는 제거된다.
    int x = 20;
    {
      int i = 20;
      System.out.println(i);
    }
    
    // 블록을 벗어났기 때문에 i 변수를 사용할 수 없다.
    //System.out.println(i); //오류!
    
    //하위 블록은 바깥 블록의 변수를 사용할 수 있다.
    //- 바깥 블록의 변수가 존재하기 때문이다.
    //int x = 30;// 같은 블록에서는 동일한 이름의 변수를 선언할 수 없다.
    {
      int i = 20;
      {
        //int i = 30; // 바깥 블록에 선언된 변수는 안쪽 블록에서 이중으로 선언할 수 없다.
        int k = 30;
        System.out.println(i);
        System.out.println(k);
        //System.out.println(m); //아직 존재하지 않는 변수는 사용할 수 없다.
        i = 100;
      }
      int m = 30;
      System.out.println(i);
      //System.out.println(k);
    }

  }

}

File: node1t/src/ex1/test06.js
var http = require('http');
var url = require('url');

// 실습 목표: 요청 URL 정보 추출하기
// - 요청과 응답에 대해 사용할 수 있는 메서드 
http.createServer(function(req, res) {
	console.log("------------------------------");
	//1) url 객체의 도움을 받아서 요청 URL을 해석하자!
	var urlInfo = url.parse(req.url);
	
	//2) 해석한 정보에서 전체 URL 정보 추출
	console.log("URL = ", urlInfo.href);
	
	//3) 해석한 정보에서 웹 애플리케이션 경로 정보 추출
	console.log("웹 애플리케이션 경로 = ", urlInfo.pathname);
	
	//4) 해석한 정보에서 웹 애플리케이션 경로를 뺀 정보 추출 
	console.log("search 스트링 = ", urlInfo.search);
	
	//5) Query String 추출
	console.log("쿼리 스트링 = ", urlInfo.query);
	
    res.writeHead(200, {
    		'Content-Type': 'text/html;charset=UTF-8',
    	});
    res.write('<html><head><title>test06</title></head>\n');
    res.write('<body><h1>test06</h1>\n');
    res.write('<p>안녕하세요!</p>\n');
    res.end('</body></html>\n');
    
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');










